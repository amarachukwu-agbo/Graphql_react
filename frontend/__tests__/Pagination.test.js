import { mount } from 'enzyme';
import wait from 'waait';
import Router from 'next/router';
import { MockedProvider } from 'react-apollo/test-utils';
import Pagination, { PAGINATION_QUERY } from '../components/Pagination';

Router.router = {
  push() {},
  prefetch() {}
}

const makeMocksFor = (length) => {
  const mocks = [
    {
      request: {
        query: PAGINATION_QUERY,
      },
      result: {
        data: {
          itemsConnection: {
            __typename: 'aggregate',
            aggregate: {
              __typename: 'count',
              count: length
            }
          }
        }
      }
    }
  ];
  return mocks;
}
describe('<Pagination />', () => {
  it('renders correctly', () => {
    mount(<MockedProvider mocks={makeMocksFor(2)}>
      <Pagination />
    </MockedProvider>);
  });

  it('displays a loading message', () => {
    const wrapper = mount(<MockedProvider mocks={makeMocksFor(2)} addTypename={false}>
      <Pagination />
    </MockedProvider>);
    expect(wrapper.find('p').text()).toBe('Loading ...');
  });

  it('render pagination for 18 items', async () => {
    const wrapper = mount(<MockedProvider mocks={makeMocksFor(18)} addTypename={false}>
      <Pagination />
    </MockedProvider>);
    await wait(0);
    wrapper.update();
    expect(wrapper.find('.totalPages').text()).toEqual('5 !')
    expect(wrapper.find('PaginationStyles')).toMatchSnapshot();
  });

  it('disables previous button on first page', async () => {
    const wrapper = mount(<MockedProvider mocks={makeMocksFor(18)} addTypename={false}>
      <Pagination page={1}/>
    </MockedProvider>);
    await wait(0);
    wrapper.update();
    expect(wrapper.find('a.prev').first().prop('aria-disabled')).toEqual(true);
  });

  it('disables next button on last page', async () => {
    const wrapper = mount(<MockedProvider mocks={makeMocksFor(18)} addTypename={false}>
      <Pagination page={5}/>
    </MockedProvider>);
    await wait(0);
    wrapper.update();
    expect(wrapper.find('a.prev').last().prop('aria-disabled')).toEqual(true);
  });

  it('enables both buttons on middle page', async () => {
    const wrapper = mount(<MockedProvider mocks={makeMocksFor(18)} addTypename={false}>
      <Pagination page={2}/>
    </MockedProvider>);
    await wait(0);
    wrapper.update();
    expect(wrapper.find('a.prev').last().prop('aria-disabled')).toEqual(false);
    expect(wrapper.find('a.prev').first().prop('aria-disabled')).toEqual(false);
  });
});