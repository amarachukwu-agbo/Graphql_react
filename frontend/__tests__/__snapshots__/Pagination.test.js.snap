// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> render pagination for 18 items 1`] = `
<PaginationStyles>
  <div
    className="PaginationStyles-aduuar-0 ewJsCc"
  >
    <SideEffect(Head)>
      <Head />
    </SideEffect(Head)>
    <Link
      href={
        Object {
          "pathname": "allitems",
          "query": Object {
            "page": NaN,
          },
        }
      }
      prefetch={true}
    >
      <a
        aria-disabled={false}
        className="prev"
        href="allitems?page="
        onClick={[Function]}
      >
        Prev
      </a>
    </Link>
    <p>
      Page 
       of 
      <span
        className="totalPages"
      >
        5
         !
      </span>
    </p>
    <p>
      18
       Items Total
    </p>
    <Link
      href={
        Object {
          "pathname": "allitems",
          "query": Object {
            "page": NaN,
          },
        }
      }
      prefetch={true}
    >
      <a
        aria-disabled={false}
        className="prev"
        href="allitems?page="
        onClick={[Function]}
      >
        Next
      </a>
    </Link>
  </div>
</PaginationStyles>
`;
